generators/chipyard-helpers/src/main/scala/chipyard-helpers/CodingStyle.scala

```diff
@@ -0,0 +1,53 @@
+class ExampleModule extends Module {
+  // val chisel stuff should be in snake case
+  val temp_wire = WireDefault(...)
+  val my_cool_reg = RegInit(...)
+
+  // scala global defaults should be all caps
+  val DEFAULT_VALUE = 20
+
+  // other scala vars should be in camelcase
+  val myOtherVar = 20
+
+  // scala functions that are associated with the module should be put in an obj
+  // notice how the function is also in camel case since it gives a scala value out
+  def myFoo(intBits: Int, chisel_val: UInt): Int = {
+
+  }
+
+  // notice how the function is also in snake case since it gives a chisel value out
+  def my_bar(intBits: Int, intBits2: Int): UInt = {
+
+  }
+}
+
+class ModuleTemplate {
+  // override this function to place all ios in a known location
```

Comment:
Can you provide an example of a module which follows this template?

Reply:
This clearly doesn't compile... but the point is something like the module below would 'implement' the template by defining the def's.

Reply:
I think we'd need to see an example of a compilable template, the syntax of the template would need to be intuitive and match how scoping in Scala and Chisel already behave.